component/block_cache/blockpool.go

```diff
@@ -109,63 +116,61 @@ func (pool *BlockPool) Terminate() {
 	pool.wg.Wait()
 
 	close(pool.blocksCh)
+	close(pool.priorityCh)
 
 	_ = pool.zeroBlock.Delete()
 
-	// Release back the memory allocated to each block
+	releaseBlock(pool.blocksCh)
+	releaseBlock(pool.priorityCh)
+}
+
+// release back the memory allocated to each block
+func releaseBlock(ch chan *Block) {
 	for {
-		b := <-pool.blocksCh
-		if b == nil {
+		block := <-ch
+		if block == nil {
 			break
 		}
-		_ = b.Delete()
+		_ = block.Delete()
 	}
 }
 
 // Usage provides % usage of this block pool
 func (pool *BlockPool) Usage() uint32 {
-	return ((pool.maxBlocks - (uint32)(len(pool.blocksCh)+len(pool.resetBlockCh))) * 100) / pool.maxBlocks
+	return ((pool.maxBlocks - (uint32)(len(pool.blocksCh)+len(pool.priorityCh))) * 100) / pool.maxBlocks
```

Comment:
We shall count the resetBlockCh as well here, as there might be blocks in that channel which are waiting to be reset